go-controller/pkg/cni/utils.go

```diff
@@ -18,22 +18,20 @@ import (
 )
 
 // wait on a certain pod annotation related condition
-type podAnnotWaitCond func(podAnnotation map[string]string) bool
+type podAnnotWaitCond func(map[string]string, string) bool
 
 // isOvnReady is a wait condition for OVN master to set pod-networks annotation
-func isOvnReady(podAnnotation map[string]string) bool {
-	if _, ok := podAnnotation[util.OvnPodAnnotationName]; ok {
-		return true
-	}
-	return false
+func isOvnReady(podAnnotation map[string]string, nadName string) bool {
+	_, err := util.UnmarshalPodAnnotation(podAnnotation, nadName)
```

Comment:
sorry, I am not sure I understand your comments. do you mean I could just check if annotation of that network already exists without un-marshaling it?

Reply:
Now, I am saying you unmarshal once before calling `isOvnReady` and then you call `isOvnReady`with the result of that. So instead of 
`func isOvnReady(podAnnotation map[string]string, nadName string) bool`
have
`func isOvnReady(podAnnotation *PodAnnotation) bool`

Reply:
sorry, I am still confused, isOvnReady is the function to keep trying to wait for the annotation of the specific nad to be ready, we cannot unmarshall beforehand, right?

Also, which other un-marshaling you are concerning? Let me know then I can think if we can optimize that.

Reply:
Annotations are un-marshaled at least twice, at least once in `isOvnReady` and once in `PodAnnotation2PodInfo`.

Other weay to do it would be to un-marshal within `isOvnReady` like now but return the result of un-marshalling instead of a boolean or return nil if not ready:
```
func isOvnReady(podAnnotation map[string]string, nadName string) *PodAnnotation
```
And then we can use the result of that in `PodAnnotation2PodInfo` without having to un-marshal again.

Just an idea, not a big issue.

Reply:
@jcaamano @trozet I've made the change, please take a look.